xbmc/peripherals/EventScanner.cpp

```diff
@@ -95,6 +95,10 @@ void CEventScanner::Process(void)
     const double nowMs = static_cast<double>(SystemClockMillis());
     const double scanIntervalMs = GetScanIntervalMs();
 
+    // Handle wrap-around
+    if (nowMs < nextScanMs)
+      nextScanMs = nowMs;
```

Comment:
Has this been tested? I don't believe floating point quantities will wrap back to zero.
They tend to stop incrementing when the exponent gets too high or reach Inf. Never zero.

Personally I think dealing with times is better done with integers.